### PR TITLE
rspec and apivore

### PR DESCRIPTION
**Pull request by rall**
This adds the rspec and apivore gems and some generated files to get started with spec writing

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -17,6 +17,8 @@ gem 'rails', '~> 5.0.0', '>= 5.0.0.1'
 group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug', platform: :mri
+  gem 'rspec-rails', '~> 3.5'
+  gem 'apivore', branch: 'master', github: 'westfieldlabs/apivore' # pull master until rails5-compatible release available
```
Comment (amcaplan) [inline Gemfile]: You should probably include these just in `test` and not `development` - no sense in slowing down the Rails console with extra gems (especially RSpec, adds around a second or so every single time!).

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# This file is copied to spec/ when you run 'rails generate rspec:install'
+ENV['RAILS_ENV'] ||= 'test'
+require File.expand_path('../../config/environment', __FILE__)
+# Prevent database truncation if the environment is production
+abort("The Rails environment is running in production mode!") if Rails.env.production?
+require 'spec_helper'
+require 'rspec/rails'
+require 'apivore'
```
Comment (amcaplan) [inline spec/rails_helper.rb]: Are you sure this line is necessary?  We don't have it in PRS...

Comment (rall) [inline spec/rails_helper.rb]: interesting paranoia from the rspec generator

Comment (amcaplan) [inline spec/rails_helper.rb]: You're saying the generator added it in? Weird.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  # This option will default to `:apply_to_host_groups` in RSpec 4 (and will
+  # have no way to turn it off -- the option exists only for backwards
+  # compatibility in RSpec 3). It causes shared context metadata to be
+  # inherited by the metadata hash of host groups and examples, rather than
+  # triggering implicit auto-inclusion in groups with matching metadata.
+  config.shared_context_metadata_behavior = :apply_to_host_groups
+
+# The settings below are suggested to provide a good initial experience
+# with RSpec, but feel free to customize to your heart's content.
+=begin
```
Comment (amcaplan) [inline spec/spec_helper.rb]: May want to uncomment this section...

**Comment by rall** [comment]
I'm merging this: it's just gems and generated files

**Pull request merged**